client/chat/channel-settings-banner.tsx

```diff
@@ -0,0 +1,302 @@
+import { debounce } from 'lodash-es'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled, { css } from 'styled-components'
+import { ReadonlyDeep } from 'type-fest'
+import { EditChannelRequest, SbChannelId } from '../../common/chat'
+import { ChannelBannerId, ChannelBannerJson } from '../../common/chat-channels/channel-banners'
+import { useForm } from '../forms/form-hook'
+import { MaterialIcon } from '../icons/material/material-icon'
+import { IconButton, useButtonState } from '../material/button'
+import Card from '../material/card'
+import { Ripple } from '../material/ripple'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { useStableCallback } from '../state-hooks'
+import {
+  blue600,
+  colorError,
+  colorTextFaint,
+  colorTextPrimary,
+  colorTextSecondary,
+} from '../styles/colors'
+import { subtitle1 } from '../styles/typography'
+import { getChannelBanners, updateChannel } from './action-creators'
+import { ChannelBanner } from './channel-banner'
+
+const ErrorText = styled.div`
+  ${subtitle1};
+  color: ${colorError};
+  margin-bottom: 8px;
+`
+
+const EmptyListText = styled.div`
+  ${subtitle1};
+  color: ${colorTextFaint};
+`
+
+const Overline = styled.div`
+  ${subtitle1};
+  color: ${colorTextSecondary};
+`
+
+const ListContainer = styled.div`
+  & + & {
+    margin-top: 32px;
+  }
+`
+
+const BannerList = styled.div`
+  padding-top: 8px;
+
+  display: flex;
+  flex-wrap: wrap;
+  gap: 8px;
+`
+
+interface ChannelSettingsBannerModel {
+  bannerId?: ChannelBannerId
+}
+
+interface ChannelSettingsBannerProps {
+  channelId: SbChannelId
+  bannerId?: ChannelBannerId
+}
+
+export function ChannelSettingsBanner({
+  channelId,
+  bannerId: originalBannerId,
+}: ChannelSettingsBannerProps) {
+  const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+  const availableBanners = useAppSelector(s => s.channelBanners.availableChannelBanners)
```

Comment:
I really do wonder whether we need to be storing these in a reducer at all, this info doesn't seem useful outside of this UI so caching it in a reducer feels unnecessary, and it complicates things significantly (since banners are particular to where you are trying to use them). Caching banner info makes sense, the rest seems like something you'd just retrieve when you need and use in the component locally